cmd/utils/flags.go

```diff
@@ -820,36 +810,18 @@ func setBootstrapNodes(ctx *cli.Context, cfg *p2p.Config) {
 			cfg.BootstrapNodes = append(cfg.BootstrapNodes, node)
 		}
 	}
-}
-
-// setBootstrapNodesV5 creates a list of bootstrap nodes from the command line
-// flags, reverting to pre-configured ones if none have been specified.
-func setBootstrapNodesV5(ctx *cli.Context, cfg *p2p.Config) {
-	urls := params.DiscoveryV5Bootnodes
-	switch {
-	case ctx.GlobalIsSet(BootnodesFlag.Name) || ctx.GlobalIsSet(BootnodesV5Flag.Name):
-		if ctx.GlobalIsSet(BootnodesV5Flag.Name) {
-			urls = splitAndTrim(ctx.GlobalString(BootnodesV5Flag.Name))
-		} else {
-			urls = splitAndTrim(ctx.GlobalString(BootnodesFlag.Name))
-		}
-	case ctx.GlobalBool(RinkebyFlag.Name):
-		urls = params.RinkebyBootnodes
-	case ctx.GlobalBool(GoerliFlag.Name):
-		urls = params.GoerliBootnodes
-	case cfg.BootstrapNodesV5 != nil:
-		return // already set, don't apply defaults.
-	}
 
-	cfg.BootstrapNodesV5 = make([]*discv5.Node, 0, len(urls))
-	for _, url := range urls {
-		if url != "" {
-			node, err := discv5.ParseNode(url)
-			if err != nil {
-				log.Error("Bootstrap URL invalid", "enode", url, "err", err)
-				continue
+	if ctx.GlobalIsSet(LegacyBootnodesV5Flag.Name) {
```

Comment:
This path will still not enter if the check exists on line 799

Reply:
If you want to merge the 2 methods, you'll need to handle the reassignment differently:

```go
if cfg.BootstrapNodes == nil {
    // Do code above
}
if cfg.BootstrapNodesV5 != nil {
    // Do code below
}
```

Reply:
Ahhh now I get it, thanks!

Reply:
fixed in https://github.com/ethereum/go-ethereum/pull/20949/commits/9e6dca0d599f6c6b97e04835ae4157a51607d2e0

Reply:
wait... why == in the first command and != in the second?

Reply:
copy paste error :) should have been == in the second too